weave/trace/vals.py

```diff
@@ -487,9 +494,25 @@ def _remote_iter(self) -> Generator[dict, None, None]:
                     if self._prefetched_rows is None
                     else self._prefetched_rows[page_index * page_size + i]
                 )
-                res = from_json(val, self.table_ref.project_id, self.server)
-                res = make_trace_obj(res, new_ref, self.server, self.root)
-                yield res
+
+                def process_row(val: Any, new_ref: RefWithExtra) -> Any:
+                    if not self.table_ref:
+                        raise ValueError("No table ref found")
+                    return make_trace_obj(
+                        from_json(val, self.table_ref.project_id, self.server),
+                        new_ref,
+                        self.server,
+                        self.root,
+                    )
+
+                future = wc.future_executor.defer(
+                    lambda v=val, r=new_ref: process_row(v, r)
+                )
+                futures.append(future)
+
+            # Yield results as they complete
+            for future in futures:
+                yield future.result()
```

Comment:
Just want to make sure this is expected:

As written, if you raise on line 500, no further futures will be returned.

Reply:
I think in practice table_ref should be defined here. i'll move that check outside the process row fn.

Reply:
oh, there already is that check at the top. ill make this less aggro but just returning none?

Reply:
hmm. I think that is what we want right? During an eval, if anything fails I believe we fail the entire thing. Additionally, all i'm doing here is a thin wrapper over the existing functionality that downloaded in the main thread, which definitely would have errored. So I think this is safe. If we in the future decided to allow partial-evals, or "resuming" evals, or fixing errored traces in evals, we might have to rethink this. Unless you can see an obvious alternative?

Reply:
ok, just want to make sure you thought about it.  That makes sense to me :)